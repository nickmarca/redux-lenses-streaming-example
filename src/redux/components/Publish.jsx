import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import classnames from 'classnames';
import { Actions as KafkaActions } from 'redux-lenses-streaming';

export class Publish extends React.Component {
  constructor(props) {
    super(props);

    this.onPublishClick = this.onPublishClick.bind(this);
    this.onInputChange = this.onInputChange.bind(this);

    this.state = {
      selectedTopic: '',
      pubKey: '',
      pubValue: '',
    };
  }

  onPublishClick() {
    const { pubKey, pubValue, selectedTopic }
      = this.state;

    const request = {
      topic: selectedTopic,
      key: pubKey,
      value: pubValue,
    };

    this.props.publish(request);
  }

  onInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    switch (name) {
      case 'selectedTopic':
      case 'pubKey':
      case 'pubValue':
        this.setState({ [name]: value });
        break;
      default:
        break;
    }
  }

  render() {
    const { connection } = this.props;
    const { pubKey, pubValue, selectedTopic }
      = this.state;

    const btnStyle = classnames('button is-fullwidth is-success');

    return (
      <nav className="panel">
        <p className="panel-heading">
          Publish Message
                </p>
        <div className="panel-block">
          <p className="control has-icons-left">
            <input
              name="pubKey"
              className="input is-small"
              type="text"
              placeholder="Key"
              value={pubKey}
              onChange={this.onInputChange}
            />
            <span className="icon is-small is-left">
              <i className="fa fa-comment" />
            </span>
          </p>
        </div>
        <div className="panel-block">
          <p className="control has-icons-left">
            <input
              name="pubValue"
              className="input is-small"
              type="text"
              placeholder="Message"
              value={pubValue}
              onChange={this.onInputChange}
            />
            <span className="icon is-small is-left">
              <i className="fa fa-comment" />
            </span>
          </p>
        </div>
        <div className="panel-block">
          <p className="control has-icons-left">
            <input
              name="selectedTopic"
              className="input is-small"
              type="text"
              placeholder="Topic"
              value={selectedTopic}
              onChange={this.onInputChange}
            />
            <span className="icon is-small is-left">
              <i className="fa fa-comment" />
            </span>
          </p>
        </div>
        <div className="panel-block">
          <button
            onClick={this.onPublishClick}
            className={btnStyle}
            disabled={!connection}
          >
            Publish
          </button>
        </div>
      </nav>
    );
  }
}

Publish.defaultProps = {
};

Publish.propTypes = {
  publish: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  connection: state.lenses.connection,
});

const mapDispatchToProps = {
  ...KafkaActions,
};

export default connect(mapStateToProps, mapDispatchToProps)(Publish);
