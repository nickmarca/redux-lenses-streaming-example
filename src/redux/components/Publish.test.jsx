import React from 'react';
import { create } from 'react-test-renderer';
import { Publish } from './Publish';

describe('Publish', () => {
    it('matches the snapshot', () => {
        const publish = create(<Publish />);

        expect(publish.toJSON()).toMatchSnapshot();
    });
});
