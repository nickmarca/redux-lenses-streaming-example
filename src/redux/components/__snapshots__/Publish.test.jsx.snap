// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish matches the snapshot 1`] = `
<nav
  className="panel"
>
  <p
    className="panel-heading"
  >
    Publish Message
  </p>
  <div
    className="panel-block"
  >
    <p
      className="control has-icons-left"
    >
      <input
        className="input is-small"
        name="pubKey"
        onChange={[Function]}
        placeholder="Key"
        type="text"
        value=""
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fa fa-comment"
        />
      </span>
    </p>
  </div>
  <div
    className="panel-block"
  >
    <p
      className="control has-icons-left"
    >
      <input
        className="input is-small"
        name="pubValue"
        onChange={[Function]}
        placeholder="Message"
        type="text"
        value=""
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fa fa-comment"
        />
      </span>
    </p>
  </div>
  <div
    className="panel-block"
  >
    <p
      className="control has-icons-left"
    >
      <input
        className="input is-small"
        name="selectedTopic"
        onChange={[Function]}
        placeholder="Topic"
        type="text"
        value=""
      />
      <span
        className="icon is-small is-left"
      >
        <i
          className="fa fa-comment"
        />
      </span>
    </p>
  </div>
  <div
    className="panel-block"
  >
    <button
      className="button is-fullwidth is-success"
      disabled={true}
      onClick={[Function]}
    >
      Publish
    </button>
  </div>
</nav>
`;
